File: HackerRank/codingClub2/b.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin >> n;
    double k = 0;
    for(int i = 0 ; i < n; ++i){
        long long int a;
        cin >> a;
        k+=(log(a));
    }
    long long int l = 1;
    long long int r = 10000000000;
    long long int ans;
    while(l <= r){
        long long int mid = (l+r)/2;
        if((n*log(mid))>k){
            ans = mid;
            r = mid-1;
        }else{
            l = mid+1;
        }
    }
    cout << ans;
    return 0;
}

File: HackerRank/codingClub2/b.py
def check(x,arr):
    f = 1
    for i in arr:
        f = f*(i/x)
    if f>=1:
        return False
    return True


n = int(input())
inp = input()
arr = [int(e) for e in inp.split()]
ans = 10000000000
l = 1
r = ans
while l<=r:
    mid = (l+r)/2
    mid = int(mid)
    if(check(mid,arr)):
        r = mid-1
        ans = mid
    else:
        l = mid+1
print(ans)